cmd/state-svc/internal/hash/file_hasher.go

```diff
@@ -32,15 +34,16 @@ type hashedFile struct {
 func NewFileHasher() *FileHasher {
 	return &FileHasher{
 		cache: cache.New(24*time.Hour, 24*time.Hour),
+		mutex: &sync.Mutex{},
 	}
 }
 
 func (fh *FileHasher) HashFiles(wd string, globs []string) (_ string, _ []hashedFile, rerr error) {
-	sort.Strings(globs) // ensure consistent ordering
+	fs := os.DirFS(wd)
 	hashedFiles := []hashedFile{}
-	hasher := xxhash.New()
+	hashes := []string{}
 	for _, glob := range globs {
-		files, err := filepath.Glob(glob)
+		files, err := doublestar.Glob(fs, glob)
```

Comment:
`filepath.Glob()` does not support double star globs (ie. `*/**`) nor specifying the working directory. This library addresses both limitations.